Add support for materialised views to postgres provider

jemrobinson opened the pull request:
Update `query_cols` SQL logic to also work with materialised views.

Closes #826

tomkralidis commented:
Thanks for the PR @jemrobinson , cc'ing additional developers for review/comment

francbartoli commented:
@jemrobinson thanks for this improvement. Can you please add some tests?

jemrobinson commented:
@francbartoli: I have added a test as requested. Note that in order to do so, I had to add a materialised view to the test PostgreSQL database. When doing this, I noticed that `hotosm_bdi_waterways.sql.gz` and `hotosm_bdi_waterways.sql` contained some small differences, so I also updated `hotosm_bdi_waterways.sql`. Let me know if this difference was deliberate and you'd like me to revert my change.
